### PR TITLE
Tutorial: Draft of Output chapter

### PR DESCRIPTION
**Pull request by killercup**

### CODE REVIEW

File: src/tutorial/output.md [hunk 1]
```diff
@@ -1,15 +1,136 @@
...
+## Showing a progress bar
+
+Some CLI applications run less than a second,
+others take minutes or hours.
+If you are writing one of the latter types of programs,
+you might want to show the user that something is happening.
+For this, you should try to printing useful status updates,
+ideally in a form that can be easily consumed.
+
+Using the [indicatif] crate,
```
Comment (spacekookie) [inline src/tutorial/output.md]: Do we only want to mention one crate here or maybe also link to https://github.com/FGRibreau/spinners (it has a `shark` spinner! :shark: )

Comment (killercup) [reply src/tutorial/output.md]: I wanted to expand this with an example that uses indicatif to show progress as well as log messages. The spinners crate can't do that afaik

File: src/tutorial/output.md [hunk 2]
```diff
@@ -1,15 +1,136 @@
...
+For example:
+
+```rust
+let x = 42;
+println!("My lucky number is {}.", x);
+```
+
+will print
+
+> My lucky number is 42.
```
Comment (artem-zinnatullin) [inline src/tutorial/output.md]: supernit: you can use ```console

So it looks like this:

```console
My lucky number is 42.
```

Comment (artem-zinnatullin) [reply src/tutorial/output.md]: Same below

File: src/tutorial/output.md [hunk 3]
```diff
@@ -1,15 +1,136 @@
...
+</aside>
+
+<aside class="todo">
+
+**TODO:**
+crate for actual log output -- which one?
+env_logger?
+
+</aside>
+
```
Comment (artem-zinnatullin) [inline src/tutorial/output.md]: Add TODO for escaping symbols in stdout/stderr?

Maybe it needs a tracking issue first, in particular I'm getting problems piping `ssh` to the stdout and stderr of my Rust program, it breaks the terminal (macOS standard Terminal.app) state quite often and I have to fix it like that:

```console
$ stty sane
```

But ideally I need to figure what symbols cause that and how to escape them…

Comment (killercup) [reply src/tutorial/output.md]: Oh, I hadn't heard of that. Can you open an issue? One goal of the issue might be to add a section to the book :)

File: src/tutorial/output.md (hunk 2 above)
Comment (killercup) [reply src/tutorial/output.md]: I <3 super nits, thanks!

**Comment by killercup** [comment]
I'll merge this and will expand it in future PRs!

File: src/tutorial/output.md (hunk 3 above)
Comment (artem-zinnatullin) [reply src/tutorial/output.md]: Sure, here we go #57

File: src/tutorial/output.md [hunk 4]
```diff
@@ -1,15 +1,140 @@
...
+## Using println
+
+You can pretty much print all the things you like
+with the `println!` macro.
+This macro has some pretty amazing capabilities,
+but also a special syntax.
+It expects you to write,
+as a first parameter,
+a literal string that with placeholders,
```
Comment (spacekookie) [inline src/tutorial/output.md]: This sentence is a bit clunky.

"It expects you to write a string literal as the first parameter, that contains placeholders that will be filled in [...]"

File: src/tutorial/output.md [hunk 5]
```diff
@@ -1,15 +1,140 @@
...
+My lucky number is 42.
+```
+
+The curly braces (`{}`) in the string above is one of these placeholders.
+This is the default placeholder type
+that tries to print the given in a human readable way.
+For numbers and strings this works very well,
+but not all types can do that.
+This is why there is also a "debug representation",
+that you can get by filling the braces of the placeholder like this: `{:?}`.
```
Comment (spacekookie) [inline src/tutorial/output.md]: There are many more od these, maybe we want to link to the section in the book where they're explained?

File: src/tutorial/output.md [hunk 6]
```diff
@@ -1,15 +1,140 @@
...
+**TODO:**
+`log` crate: macros with similar syntax to `println`
+
+</aside>
+
+<aside class="todo">
+
+**TODO:**
+crate for actual log output -- which one?
+env_logger?
```
Comment (spacekookie) [inline src/tutorial/output.md]: We might want to include several and compare? The book is supposed to be curated content, right? So maybe keeping a list of actively developed loggers isn't a bad idea

File: src/tutorial/output.md (hunk 5 above)
Comment (killercup) [reply src/tutorial/output.md]: there's a link to the std::fmt docs in the very next aside

File: src/tutorial/output.md (hunk 6 above)
Comment (killercup) [reply src/tutorial/output.md]: Right, but this is also the tutorial. I'll add a todo to link to `../in-depth/human-communication.html` :)

**Pull request merged**